test/unit/reductions/grid_search/test_grid_generator.py

```diff
@@ -34,6 +34,34 @@ def test_grid_generator_demographic_parity(grid_size, grid_limit):
     assert np.isclose(expected_grid.values, grid.values).all()
 
 
+@pytest.mark.parametrize("grid_size", [5, 6])
+@pytest.mark.parametrize("grid_limit", [1.0, 2.0])
+@pytest.mark.parametrize("grid_offset", [[0, 0, 0, 0], [0, 0, 0, 1], [0, 1, 0, 0],
```

Comment:
does the case with 0 0 0 0 not cover the entire test case above?

Reply:
Yes. It does. I'll remove it from here.

Reply:
Oh no I meant remove the test case above and just cover it through this much more general test case! Sorry, should have been more explicit :-)

Reply:
I misunderstood this earlier. I have fixed it now. This PR is complete from my side.